frontend/components/color_encoding_to_sound/BasePage.js

```diff
@@ -2,63 +2,123 @@ import React from "react";
 import {SketchPicker} from 'react-color';
 import {fetchPost} from "../../common";
 import SliderInstrument from "../instruments/SliderInstrument";
+import PaletteColor from "../color_encoding_to_sound/PaletteColor";
 
 class BasePage extends React.Component {
-    state = {
-        color: {
-            r: 51,
-            g: 51,
-            b: 51,
-        },
-        result: [],
-        listOfColors: [],
-    };
+    constructor(props) {
+        super(props);
+
+        this.handlePaletteClick = this.handlePaletteClick.bind(this);
+        this.state = {
+            result: [],
+            selected: 0,
+            instrumentGenerated: false,
+            colorPickerColor: {
+                    r: 51,
+                    g: 51,
+                    b: 51,
+            },
+            listOfColors: [
```

Comment:
Usually I'm in favor of writing things like this out explicitly, but this is a situation in which generating the initial data dynamically might make what's going on easier to read. Something like...

```
const NUM_COLORS = 7;
const initialListOfColors = []
for (let i = 0; i < NUM_COLORS; i++) {
    const black = {r: 0, g: 0, b:0}
    initialListOfColors.push(black);
}
this.state = {
    listOfColors: initialListOfColors
   // ... other stuff...
}
```

Also means we can factor out the number of colors in the palette and iterate on that.